src/data-connect/data-connect-api-client-internal.ts

```diff
@@ -0,0 +1,233 @@
+/*!
+ * @license
+ * Copyright 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { App } from '../app';
+import { FirebaseApp } from '../app/firebase-app';
+import {
+  HttpRequestConfig, HttpClient, RequestResponseError, AuthorizedHttpClient
+} from '../utils/api-request';
+import { PrefixedFirebaseError } from '../utils/error';
+import * as utils from '../utils/index';
+import * as validator from '../utils/validator';
+import { ConnectorConfig, ExecuteGraphqlResponse, GraphqlOptions } from './data-connect-api';
+
+// Data Connect backend constants
+const DATA_CONNECT_HOST = 'https://firebasedataconnect.googleapis.com';
+//const DATA_CONNECT_EMULATOR_HOST = 'http://127.0.0.1:9399';
+const DATA_CONNECT_API_URL_FORMAT =
+  '{host}/v1alpha/projects/{projectId}/locations/{locationId}/services/{serviceId}:{endpointId}';
+
+const EXECUTE_GRAPH_QL_ENDPOINT = 'executeGraphql';
+const EXECUTE_GRAPH_QL_READ_ENDPOINT = 'executeGraphqlRead';
+
+const DATA_CONNECT_CONFIG_HEADERS = {
+  'X-Firebase-Client': `fire-admin-node/${utils.getSdkVersion()}`
+};
+
+/**
+ * Class that facilitates sending requests to the Firebase Data Connect backend API.
+ *
+ * @internal
+ */
+export class DataConnectApiClient {
+  private readonly httpClient: HttpClient;
+  private projectId?: string;
+
+  constructor(private readonly connectorConfig: ConnectorConfig, private readonly app: App) {
+    if (!validator.isNonNullObject(app) || !('options' in app)) {
+      throw new FirebaseDataConnectError(
+        'invalid-argument',
```

Comment:
Suggest using the map here: `DATA_CONNECT_ERROR_CODE_MAPPING.INVALID_ARGUMENT`? and in other locations, to reduce chance of typos.

Reply:
Good call! Thanks!